### PR TITLE
Deploy contracts for smoke tests

### LINKED ISSUE #1645: Smoke tests should deploy the contracts and not use precompiled ones
Since the smoketest is supposed to be the minimal integration tests we have and will always be running, it should be deploying the contracts on its own and not use precompiled contracts stored in the genesis file.

When we do this we can also get rid of the [hacky patch function](https://github.com/raiden-network/raiden/blob/da745f5eff87f4e81ed7b2be5a2ff5cc1d76c359/raiden/tests/utils/smoketest.py#L234) to make the smoketests work with events querying.

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #1645

### CODE REVIEW

File: raiden/utils/deployment.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    'ChannelManagerLibrary.sol',
+    'Registry.sol',
+    'EndpointRegistry.sol',
+]
+
+# Top level contracts to be deployed. Dependencies are handled automatically
+# in `JSONRPCClient.deploy_solidity_contract()`
+CONTRACTS_TO_DEPLOY = [
+    'Registry.sol:Registry',
+    'EndpointRegistry.sol:EndpointRegistry',
```
Comment (hackaugusto) [inline raiden/utils/deployment.py]: what about adding the secret registry here?

File: raiden/utils/deployment.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        '',
+        contract_path=filename,
+    )
+
+    log.info(f"Deployed {name} @ {to_checksum_address(proxy.contract_address)}")
+    libraries[name] = proxy.contract_address
+    return libraries
+
+
+def deploy_contracts(client):
```
Comment (hackaugusto) [inline raiden/utils/deployment.py]: shouldn't we make the contracts an argument of this function?

**Comment by LefterisJP** [comment]
@rakanalh we don't add pull requests to milestones

File: raiden/utils/deployment.py (hunk 1 above)
Comment (rakanalh) [reply raiden/utils/deployment.py]: Added below this list as a separate list which would ease cleanup once the new contracts are fully integrated and old ones are removed.

**Pull request merged**